components/tidb_query_datatype/src/codec/mysql/time/mod.rs

```diff
@@ -2714,6 +2721,20 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_from_local_time() -> Result<()> {
+        let mut ctx = EvalContext::default();
+        for i in 2..10 {
+            let actual = Time::from_local_time(&mut ctx, TimeType::DateTime, i % MAX_FSP)?;
+            let c_datetime = actual.try_into_chrono_datetime(&mut ctx)?;
+
+            let now0 = &c_datetime.to_string()[0..22]; //2 meas milli_seconds/10 length
```

Comment:
Why `..22` here? You may want to use `[..]`?

Reply:
"c_datetime" and "Utc::now()" as string will be:
2022-02-15 07:45:48.581699 NonFixed(UTC)
2022-02-15 07:45:48.581933075 UTC

We only compare length 22 of then are enough; Exceed it my be not equal value ( in milli seconds );
so be [0..22]:
let now0 = &c_datetime.to_string()[0..22]; 
let now1 = &Utc::now().to_string()[0..22];

Reply:
Would you please add the above comments to the Rust code as comments?

Reply:
ok